lib/src/sql/value/get.rs

```diff
@@ -222,8 +222,14 @@ impl Value {
 						},
 					}
 				}
-				// Ignore everything else
-				_ => Ok(Value::None),
+				v => {
+					if matches!(p, Part::Flatten) {
+						v.get(ctx, opt, txn, None, path.next()).await
+					} else {
+						// Ignore everything else
+						Ok(Value::None)
+					}
+				}
```

Comment:
Because, in this case, we come from the Object (not anymore from the Array).
As `Part::Flatten` it is another step in the idiom. It covers this case:
`emails.*.value[+]`